.github/workflows/tarball.yml

```diff
@@ -0,0 +1,247 @@
+        with:
+          name: linux-stanc
+          path: linux-stanc
+
+  windows:
+    runs-on: ubuntu-latest
+    container:
+      image: ubuntu:bionic
+      options: --user root
+    steps:    
+      - name: Checkout stanc3
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.event.inputs.stanc3_repo }}
+          ref: ${{ github.event.inputs.stanc3_branch }}
+
+      - name: opam caching
+        id: opam-cache
+        uses: actions/cache@v2
+        with:
+          path: "~/.opam"
+          key: windows-stanc-binary-4.07
+
+      - name: Install dependencies
+        run: |
+          apt-get update -qq
+          apt-get install -y --no-install-recommends ca-certificates rsync git build-essential m4 sudo \
+          unzip pkg-config libpcre3-dev mingw-w64 gcc wget gawk bubblewrap
+          wget https://github.com/ocaml/opam/releases/download/2.0.4/opam-2.0.4-x86_64-linux
+          sudo install opam-2.0.4-x86_64-linux /usr/local/bin/opam
+          opam init --comp 4.02.3 --disable-sandboxing -y
+          eval $(opam env)
+          opam switch 4.07.0 || opam switch create 4.07.0
+          eval $(opam env)
+          opam update
+          bash -x scripts/install_build_deps_windows.sh
+          eval $(opam env)
+          opam install -y js_of_ocaml-compiler.3.4.0 js_of_ocaml-ppx.3.4.0 js_of_ocaml.3.4.0
+
+      - name: Build Windows binaries
+        run: |
+          eval $(opam env)
+          dune subst
+          dune build -x windows
+          mv _build/default.windows/src/stanc/stanc.exe windows-stanc
+
+      - name: Upload Windows stanc
+        uses: actions/upload-artifact@v2
+        with:
+          name: windows-stanc
+          path: windows-stanc
+  mac:
+    runs-on: macOS-latest
+    steps:    
+      - name: Checkout stanc3
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.event.inputs.stanc3_repo }}
+          ref: ${{ github.event.inputs.stanc3_branch }}
+
+      - name: opam caching
+        id: opam-cache
+        uses: actions/cache@v2
+        with:
+          path: "~/.opam"
+          key: macOS-stanc-binary-4.07
+
+      - name: Install dependencies
+        run: |
+          brew install gpatch
+          brew install opam
+          opam init --disable-sandboxing -y
+          eval $(opam env)
+          opam switch 4.07.0 || opam switch create 4.07.0          
+          eval $(opam env)
+          bash -x scripts/install_build_deps.sh
+
+      - name: Build MacOS binaries
+        run: |
+          eval $(opam env)
+          dune subst
+          dune build @install
+          mv _build/default/src/stanc/stanc.exe mac-stanc
+      
+      - name: Upload MacOS binaries
+        uses: actions/upload-artifact@v2
+        with:
+          name: mac-stanc
+          path: mac-stanc
+  build:
+    needs:  [linux, windows, mac]
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/setup-python@v1
```

Comment:
@rok-cesnovar would be nice to use precise Python specifications like version and architecture.

See the [`setup-python` docs](https://github.com/actions/setup-python#usage)

```yaml
- uses: actions/setup-python@v2
  with:
    python-version: '3.x' # Version range or exact version of a Python version to use, using SemVer's version range syntax
    architecture: 'x64' # optional x64 or x86. Defaults to x64 if not specified
```

Reply:
Good call. Thanks!